code/game/objects/items/rcd/RPD.dm

```diff
@@ -257,6 +278,10 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 
 /obj/item/pipe_dispenser/dropped(mob/user, silent)
 	UnregisterSignal(user, COMSIG_MOUSE_SCROLL_ON)
+	if(!(upgrade_flags & RPD_UPGRADE_SPEEDPIPE))
+		return .
```

Comment:
`return .` doesn't do anything here because you don't set it to the parent value